### PR TITLE
RedisStore: Refactoring of sample endpoints - get, find and upsert

### PR DESCRIPTION
**Pull request by pallavi2209**
Refactored get and find sample, removed apilinks from common method. Set links for find samples. Refactored upsert sample to not to pass req.method.

### CODE REVIEW

File: api/v1/controllers/samples.js [hunk 1]
```diff
@@ -83,27 +81,7 @@ module.exports = {
...
-        }
-
-        u.logAPI(req, resultObj, sampleRes); // audit log
-        res.status(httpStatus.OK).json(sampleRes);
-      })
-      .catch((err) => u.handleError(next, err, helper.modelName));
-    } else {
-      doGet(req, res, next, helper);
-    }
+    doGet(req, res, next, helper);
```
Comment (annyhe) [inline api/v1/controllers/samples.js]: yay less code :)

File: api/v1/controllers/samples.js [hunk 2]
```diff
@@ -172,9 +150,7 @@ module.exports = {
           u.checkDuplicateRLinks(sampleQueryBody.relatedLinks);
         }
 
-        upsertSamplePromise = redisModelSample.upsertSample(
-          sampleQueryBody, resultObj, res.method
-        );
+        upsertSamplePromise = redisModelSample.upsertSample(sampleQueryBody);
       } else {
```
Comment (annyhe) [inline api/v1/controllers/samples.js]: are we still logging in redisModelSample.upsertSample?

File: cache/models/samples.js [hunk 3]
```diff
@@ -777,7 +766,6 @@ module.exports = {
 
     return redisClient.batch(commands).execAsync()
     .then((responses) => {
-      logObject.dbTime = new Date() - logObject.reqStartTime; // log db time
       const sample = responses[ZERO];
```
Comment (annyhe) [inline cache/models/samples.js]: did we remove logging for this endpoint?

File: api/v1/controllers/samples.js (hunk 2 above)
Comment (pallavi2209) [inline api/v1/controllers/samples.js]: Logging before returning response. Check the code after the changes.

File: cache/models/samples.js (hunk 3 above)
Comment (pallavi2209) [inline cache/models/samples.js]: Logging in doGet verb instead.

File: api/v1/helpers/verbs/doGet.js [hunk 4]
```diff
@@ -65,17 +68,18 @@ function doGet(req, res, next, props) {
...
-      // loop through remove values to delete property
-      if (props.fieldsToExclude) {
-        u.removeFieldsFromResponse(props.fieldsToExclude, o.dataValues);
-      }
+    getPromise.then((o) => {
+      resultObj.dbTime = new Date() - resultObj.reqStartTime;
 
-      // o is a sequelize obj, get dataValues obj
-      u.logAPI(req, resultObj, o.dataValues);
+      u.logAPI(req, resultObj, o);
```
Comment (annyhe) [inline api/v1/helpers/verbs/doGet.js]: what is o? is the logged object the dataValues object, ie. the API response?

Comment (pallavi2209) [inline api/v1/helpers/verbs/doGet.js]: o can be sequelize object or redis object, not datavalues object.

Comment (annyhe) [inline api/v1/helpers/verbs/doGet.js]: o is used by the logging utility to return the approximate the size of returned information. If o is the sequelize object or redis object, the approximation may be way off. 
We may need to modify the getSize(object) function in utils/apiLog.js to parse the object.

Comment (pallavi2209) [inline api/v1/helpers/verbs/doGet.js]: I extract the dataValues from Sequelize object in apiLog function. Please  look into details to confirm.

**Pull request merged**